src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.cs

```diff
@@ -324,10 +333,32 @@ private static void OnItemsRepeaterElementPrepared(ItemsRepeater sender, Microso
 
 		SetItemSelection(sender, args.Element, selected);
 	}
+
 	private static void OnItemsRepeaterItemsSourceChanged(DependencyObject sender, DependencyProperty dp)
 	{
 		// When we reach here, ItemsSourceView is already updated.
 		if (sender is not ItemsRepeater ir) return;
+
+#if HAS_UNO_WINUI
+		// Except for WASM
+		ir.ItemsSourceView.CollectionChanged += OnItemsSourceViewCollectionChanged;
```

Comment:
Binding the event to the callback was causing silent crashes when I tried doing it here https://github.com/unoplatform/uno.toolkit.ui/blob/3d2632af64d41991fdf9ff68fcb8bfda1a8e2a79/src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.cs#L201-L210

Also, what's the best way to dispose of it? Using Disposable.Create() like here? https://github.com/unoplatform/uno.toolkit.ui/blob/3d2632af64d41991fdf9ff68fcb8bfda1a8e2a79/src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.cs#L205-L209

Reply:
I'm not sure I understand the question, we should be disposing the CompositeDisposable here, right?

https://github.com/unoplatform/uno.toolkit.ui/blob/3d2632af64d41991fdf9ff68fcb8bfda1a8e2a79/src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.cs#L231

Reply:
the crash is normal, since ItemsSourceView is update on when ItemsSource is
likely at the time of this extensions being set, that could be null still

Reply:
you should:
- create a seperate SerialDisposable for this subscription
- add that serial-disp to the composite-disp
- and clear then set (dont just re-assign) the inner disposable here: (_SourceViewCollectionChangedDisposable as SerialDisposable).Disposable = Disposable.Create(() => ir.ItemsSourceView.CollectionChanged -= OnItemsSourceViewCollectionChanged)